### ISSUE TITLE
Potential bug in plotting a simple line

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#6217]
**Full script that generated the error**

```
gmt plot -R0/5/0/5 -JX8c -Ba1g1 -W2p,blue -pdf test << EOF
2 4
2 3
2 1
2 3
EOF
```

**Actual outcome**

![test](https://user-images.githubusercontent.com/3974108/149125140-3ebc475b-e8ff-4283-a892-a3a08faec9b0.png)

I expect to see a line between (2,3) and (2,1), but I don't see it. Is it a bug or am I making a stupid mistake?

**System information**

* Operating system: Linux 
* GMT version (`gmt --version`): GMT 6.3.0

**Comment by Esteban82** [comment]
I have the same map (with GMT dev in Linux).

I even modified the data a bit. The line is deleted when is superimposed. If 
```
gmt plot -R0/5/0/5 -JX8c -Ba1g1 -W2p,blue -pdf test << EOF
2 4
2 1
2 3.5
EOF

```
![test](https://user-images.githubusercontent.com/17908951/149140776-747d1019-273e-4636-b61f-9eefd7a463c6.png)

Here I use `2.01 3.5` for the last point. 

![test](https://user-images.githubusercontent.com/17908951/149141486-f9bb7c6a-083f-438b-82fa-350bf295dcfe.png)

**Comment by Esteban82** [comment]
I think it was fixed by #6234.
